homeassistant/components/mastodon/notify.py

```diff
@@ -36,38 +44,73 @@
 )
 
 
-def get_service(
+async def async_get_service(
     hass: HomeAssistant,
     config: ConfigType,
     discovery_info: DiscoveryInfoType | None = None,
 ) -> MastodonNotificationService | None:
     """Get the Mastodon notification service."""
-    client_id = config.get(CONF_CLIENT_ID)
-    client_secret = config.get(CONF_CLIENT_SECRET)
-    access_token = config.get(CONF_ACCESS_TOKEN)
-    base_url = config.get(CONF_BASE_URL)
-
-    try:
-        mastodon = Mastodon(
-            client_id=client_id,
-            client_secret=client_secret,
-            access_token=access_token,
-            api_base_url=base_url,
+
+    if not discovery_info:
+        # Import config entry
+        client_id = config.get(CONF_CLIENT_ID)
+        client_secret = config.get(CONF_CLIENT_SECRET)
+        access_token = config.get(CONF_ACCESS_TOKEN)
+        base_url = config.get(CONF_BASE_URL, DEFAULT_URL)
+        name = config.get(CONF_NAME, "Mastodon")
+
+        import_result = await hass.config_entries.flow.async_init(
+            DOMAIN,
+            context={"source": SOURCE_IMPORT},
+            data={
+                CONF_CLIENT_ID: client_id,
+                CONF_CLIENT_SECRET: client_secret,
+                CONF_ACCESS_TOKEN: access_token,
+                CONF_NAME: name,
+                CONF_BASE_URL: base_url,
+            },
+        )
+
+        translation_key = "deprecated_yaml"
+        if import_result.get("type") == FlowResultType.ABORT:
+            translation_key = "import_aborted"
+            if import_result.get("reason") == "import_failed":
+                translation_key = "import_failed"
+
+        async_create_issue(
+            hass,
+            DOMAIN,
+            f"config_import_{DOMAIN}",
+            breaks_in_ha_version="2025.2.0",
+            is_fixable=False,
+            is_persistent=False,
+            issue_domain=DOMAIN,
+            severity=IssueSeverity.WARNING,
+            translation_key=translation_key,
+            translation_placeholders={
+                "domain": DOMAIN,
+                "integration_title": "Mastodon",
+            },
         )
```

Comment:
If the entry is already imported, or the abort succeeded, we should raise the generic Home Assistant issue, I think incomfort has an example for that.